.github/workflows/beam_PerformanceTests_Cdap.yml

```diff
@@ -0,0 +1,108 @@
+  cancel-in-progress: true
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+  beam_PerformanceTests_Cdap_test_arguments_1: ''
+
+jobs:
+  beam_PerformanceTests_Cdap:
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Java CdapIO Performance Test'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 100
+    name: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    strategy:
+      matrix:
+        job_name: ["beam_PerformanceTests_Cdap"]
+        job_phrase: ["Run Java CdapIO Performance Test"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          comment_phrase: ${{ matrix.job_phrase }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+      - name: Authenticate on GCP
+        id: auth
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: ${{ secrets.GCP_SA_KEY }}
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+      - name: Set k8s access
+        uses: ./.github/actions/setup-k8s-access
+        with:
+          cluster_name: io-datastores
+          k8s_namespace: ${{ matrix.job_name }}-${{ github.run_id }}
+      - name: Install postgres
+        id: install_postgres
+        run: |
+          kubectl apply -f ${{ github.workspace }}/.test-infra/kubernetes/postgres/postgres-service-for-local-dev.yml
+          kubectl wait svc/postgres-for-dev --for=jsonpath='{.status.loadBalancer.ingress[0].ip}' --timeout=60s
+          loadbalancer_IP=$(kubectl get svc postgres-for-dev -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+          echo postgres_IP=$loadbalancer_IP >> $GITHUB_OUTPUT
+      - name: Prepare test arguments
+        uses: ./.github/actions/test-arguments-action
+        with:
+          test-type: performance
+          test-language: java
+          argument-file-paths: |
+            ${{ github.workspace }}/.github/workflows/performance-tests-job-configs/cdap.txt
+          arguments: |
+            --postgresServerName=${{ steps.install_postgres.outputs.postgres_IP }}
+      - name: run integrationTest
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+          gradle-command: :sdks:java:io:cdap:integrationTest
+          arguments: |
+            --info \
+            --tests org.apache.beam.sdk.io.cdap.CdapIOIT \
+            -DintegrationTestRunner=dataflow \
+            -DintegrationTestPipelineOptions='[${{ env.beam_PerformanceTests_Cdap_test_arguments_1 }}]' \
```

Comment:
There are a couple of leftover workloads likely due to manual run. 

<img width="430" alt="image" src="https://github.com/apache/beam/assets/8010435/c3aa61b1-788c-4dff-8d83-11b12ef15de9">

and I do not see a teardown step in these workflows. Please check

Reply:
note that some are due to https://github.com/apache/beam/blob/master/.github/workflows/beam_PerformanceTests_AvroIOIT_HDFS.yml on master, not this PR though

Reply:
@Abacn Teardown is in the setup-k8s-access action and needs approval for

pytooling/actions/with-post-step@v0.4.6

Reply:
@Abacn @volatilemolotov These namespaces are from both Jenkins and GitHub Actions - the namespaces which have a short build numbers (e.g. `beam-python-mongoio-load-test-5071`) are Jenkins ones and the long build numbers are related to GH Actions (e.g. `beam-performancetests-avroioit-hdfs-6303887955`). The GH Actions namespaces are listed because the `Post Set k8s access` step was not triggered due to the cancelled run in our forks:
- https://github.com/andreydevyatkin/beam/actions/runs/6303887955
- https://github.com/andreydevyatkin/beam/actions/runs/6324129155

At this moment, we are waiting for approval from the infra team to whitelist the `setup-k8s-access` action https://github.com/apache/beam/pull/28530#issuecomment-1734560209

Reply:
Thanks for clarification